wherehows-dao/src/main/java/wherehows/dao/table/SearchDao.java

```diff
@@ -34,20 +34,20 @@
 
   private static final ObjectMapper _OM = new ObjectMapper();
 
-  public List<String> getAutoCompleteList(String elasticSearchUrl, String input, int limit) {
-    return getAutoCompleteListDataset(elasticSearchUrl, input, limit / 3);
+  public List<String> getAutoCompleteList(String elasticSearchUrl, String input, int limit, String facet) {
+    return getAutoCompleteListDataset(elasticSearchUrl, input, limit / 3, facet);
   }
 
-  public List<String> getAutoCompleteListDataset(String elasticSearchUrl, String input, int limit) {
-    String fieldName = "name_suggest";
-    return getAutoCompleteListbyES(elasticSearchUrl, fieldName, input, limit);
+  public List<String> getAutoCompleteListDataset(String elasticSearchUrl, String input, int limit, String facet) {
+    return getAutoCompleteListbyES(elasticSearchUrl, facet, input, limit);
   }
 
   public List<String> getAutoCompleteListbyES(String elasticSearchUrl, String fieldName, String input, int limit) {
     // use elastic search completion suggester, ES will validate the input and limit
     List<String> completionSuggestionList = new ArrayList<>();
     Set<String> completionSuggestionSet = new HashSet<>();
 
+    fieldName = "name_suggest";
```

Comment:
Why are we setting the field name when it's already passed as an input?

Reply:
in os searchdao, the value is overwritten to be backward compatible as it only uses name_suggest.